Homepage: add overview of full node benefits

harding opened the pull request:
Preview: http://dg1.dtrt.org/

This adds to the homepage a short one-paragraph overview of what a full node does.  A button designed to look like a link can be clicked to display an additional four paragraphs about the benefits of Bitcoin Core.  Two prominent links are emphasized pointing to pages I'd expect readers to find especially useful.

![2021-06-10-12_09_05_857152768](https://user-images.githubusercontent.com/61096/121603685-db114280-c9e4-11eb-9218-04453c9cbd2b.png)

This is the first part of a series of PRs designed to copy most of the content from the old Bitcoin.org Bitcoin Core [subsite](https://web.archive.org/web/20150930013906/https://bitcoin.org/en/bitcoin-core/) to this website, updating it in the process.  When all that content has been copied, two further changes are expected to this page:

1. The final paragraph of the extended description, which currently contains no links, will be updated to link various words to the corresponding feature pages.

2. The two icon-title links below the text (Download and Contribute) will be joined by additional links to Features and Get Help.

Two small commits in this PR affect other parts of the site:

- "CSS: change..." updates the default color for hyperlinks from black to blue.  The black text on the existing site often looks to me like bolding rather than something you can click, and I think that needs to be fixed if we want to guide users to a larger set of available resources, like various pages with descriptions of Bitcoin Core's features.

- "Downloads: refactor..." takes code we have for finding the highest-number release on the site and makes it a function we can use on other pages, allowing us to mention the latest version of Bitcoin Core on the homepage.  I could've done this as a move-only but the code, which I originally wrote, was dumb, so I made it less dumb (hopefully).

The new text will not appear on the translated homepages unless the translators update their strings and bump the version numbers.

harding commented:
Thanks to @hebasto for poking me to put more resources on the website that educate users about the need for, and benefits of, economically using your own full nodes.  Also many thanks to those who contributed to and helped review the original PR for this on Bitcoin.org almost six years ago: https://github.com/bitcoin-dot-org/Bitcoin.org/pull/1044

The next step in this series of PRs will be adding an updated version of the [validation page](https://web.archive.org/web/20150929204455/https://bitcoin.org/en/bitcoin-core/features/validation) (which I think I'll use this opportunity to rename to *verification*).

hebasto commented:
Concept ACK.

hebasto reviewed (approved):
ACK dd5aa3d486a3f21e61294e852eee282c5fca55e9.

Hope site improvements like this one will induce Bitcoin Core users to educate themselves about distinguished benefits for their privacy and their funds safety.

harding commented:
Realized this was accidentally based on an old branch and was showing 0.21.0 as the latest release; rebased to master with no code changes and confirmed it now correctly shows 0.21.1 as the latest release.  Also updated the preview.  Sorry for the confusion.

hebasto reviewed (approved):
re-ACK 98889f9ce08025de86a6bc086c4de8ab941396d5, only rebased since my [previous](https://github.com/bitcoin-core/bitcoincore.org/pull/783#pullrequestreview-681338184) review, verified with
```
git range-diff master dd5aa3d486a3f21e61294e852eee282c5fca55e9 98889f9ce08025de86a6bc086c4de8ab941396d5
```

Rspigler commented on _posts/en/pages/2016-01-01-index.md:
```diff
@@ -5,9 +5,44 @@ lang: en
+about2: >
+  It is these users who keep Bitcoin decentralized. They
+  individually run their own Bitcoin Core full nodes, and each of
+  those full nodes separately follows the exact same rules to decide
```
```suggestion
  those full nodes separately follow the exact same rules to decide
```

harding replied:
I believe the agreement here is correct.

Rspigler replied:
Full nodes follow
Full node follows.

"follow" should be singular

harding replied:
It still sounds wrong to me in the singular, but I don't think it's important, so I'll change it.

Rspigler commented on _posts/en/pages/2016-01-01-index.md:
```diff
@@ -5,9 +5,44 @@ lang: en
+  which block chain is valid.
+
+about3: >
+  There's no voting or other corruptible process involved: there's
+  just individual software following identical rules to evaluate
+  identical blocks and coming to identical conclusions about which block
+  chain is valid.
```
```suggestion
  There's no voting or other corruptible process involved: there's
  just individual software following identical rules to evaluate
  data in order to come to identical conclusions about which block
  chain is valid.
```

harding replied:
What's the nit here?

Rspigler replied:
It's better (IMO) when the verbs are in the same form (to evaluate, to come).  Otherwise, since "and coming to identical conclusions about which block chain is valid" is not a full sentence, you need a comma before the "and".

I also changed "identical blocks" -> "data" because full nodes are evaluated blocks are transactions, and because they could be fed different transactions, and I guess technically different blocks by attackers.

harding replied:
I don't understand:

> I also changed "identical blocks" -> "data" because full nodes are evaluated blocks are transactions, and because they could be fed different transactions, and I guess technically different blocks by attackers.

Rspigler replied:
Woops, sorry.

I also changed "identical blocks" -> "data" because full nodes evaluate blocks and transactions, and also because nodes could be fed different transactions, and I guess technically different blocks by attackers.

Rspigler reviewed (commented):
Strong concept ack! Few grammar nits

Zero-1729 reviewed (approved):
ACK ab307e3dfb94c044cbfe4f0fc3204c6df0fe82b9

Glad to see additions like these! I hope it gets more folks educated about full nodes and excited to spin them up 📡

prayank23 commented on _posts/en/pages/2016-01-01-index.md:
```diff
@@ -5,9 +5,44 @@ lang: en
+
+about4: >
+  This shared agreement (called consensus) allows people like you to
+  only accept valid bitcoins, enforcing Bitcoin's rules without
```
```suggestion
  only accept valid bitcoin, enforcing Bitcoin's rules without
```

prayank23 commented on _posts/en/pages/2016-01-01-index.md:
```diff
@@ -5,9 +5,44 @@ lang: en
+
+about5: >
+  In addition to improving Bitcoin's decentralization, Bitcoin Core
+  users get better security for their bitcoins, privacy features not
```
```suggestion
  users get better security for their bitcoin, privacy features not
```

katesalazar commented:
- [x] concept ack
- [ ] tested
